File: Lab3OOP/Song.cpp
#include "Song.h"
#include "iostream"
#include <sstream>
Song::Song(const std::string& title, int duration, const std::string& vocalist) : Composition(title, duration), vocalist(vocalist) {}

std::string Song::show() const {
    std::ostringstream oss;
    oss << "Title: " << title << ", Duration: " << duration << " sec, Vocalist: " << vocalist;
    return oss.str();
}

std::string Song::getVocalist() const {
    return vocalist;
}

File: Lab3_oop_python/song.py
from composition import Composition

class Song(Composition):
    def __init__(self, title: str, duration: int, vocalist: str):
        super().__init__(title, duration)
        self.vocalist = vocalist

    def get_vocalist(self) -> str:
        return self.vocalist
    def show(self):
       return(f"Title: {self.title}, Duration: {self.duration} sec, Vocalist: {self.vocalist}")
